src/datasets/arrow_dataset.py

```diff
@@ -815,20 +823,51 @@ def cast_(self, features: Features):
                 The name of the fields in the features must match the current column names.
                 The type of the data must also be convertible from one type to the other.
                 For non-trivial conversion, e.g. string <-> ClassLabel you should use :func:`map` to update the Dataset.
+            batch_size (`Optional[int]`, defaults to `1000`): Number of examples per batch provided to `function` if `batched=True`
```

Comment:
Casting the tables in batched way is faster than doing the cast row by row.
Therefore I didn't add the `batched` parameter here, since there's no point for users to cast in a not-batched way.

Reply:
Yes, I totally agree with your implementation. What I wanted to say is that it is pointless to mention the parameter `batched` in the docstring (and also the parameter `function`, BTW; I guess these were just copied from the `map` docstring). 

I would rather write something like:
```
Number of examples per batch provided to cast.
```

Why?
- Parameters `batched` and `function` do not appear in the signature of the method.
- Those parameters are internal local variables (private). 
- Those parameter values cannot be changed: `batched=True` and `function=lambda t: t.cast(schema)`.

Reply:
Yes right ! I'm changing that

Reply:
done :)